tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -2605,6 +2606,219 @@ public static void test_supportsLikeEscapeClause_shouldBe_true() throws Exceptio
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next(), "Expected exactly 1 row from information_schema.tables, got 0");
+			assertFalse(resultSet.next());
+			resultSet.close();
+
+			try {
+				statement.execute("DROP TABLE " + qualifiedTableName);
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInTableDefinitions) {
+					fail(
+						"supportsCatalogsInTableDefinitions is true but DROP TABLE in attached database is not allowed. " + 
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next() == false, "Expected exactly 0 rows from information_schema.tables, got > 0");
+			resultSet.close();
+
+			assertTrue(supportsCatalogsInTableDefinitions, "supportsCatalogsInTableDefinitions should return true.");
+		} 
+		finally {
+			if (resultSet != null) {
+				resultSet.close();
+			}
+			statement.close();
+			connection.close();
+		}
+	}
+
+	public static void test_supports_catalogs_in_data_manipulation() throws Exception {
+		final String CATALOG_NAME = "tmp";
+		final String TABLE_NAME = "t1";
+		final String COLUMN_NAME = "id";
+		final String qualifiedTableName = CATALOG_NAME + "." + TABLE_NAME;
+
+		final Connection connection = DriverManager.getConnection("jdbc:duckdb:");
+		final DatabaseMetaData databaseMetaData = connection.getMetaData();
+		final Statement statement = connection.createStatement();
+		ResultSet resultSet = null;
+		try {
+			statement.execute("ATTACH '' AS \"" + CATALOG_NAME + "\"");
+			statement.execute("CREATE TABLE " + qualifiedTableName + "(" + COLUMN_NAME + " int)");
+			
+			final boolean supportsCatalogsInDataManipulation = databaseMetaData.supportsCatalogsInDataManipulation();
+			try {
+				statement.execute("INSERT INTO " + qualifiedTableName + " VALUES(1)");
+				resultSet = statement.executeQuery("SELECT * FROM "+ qualifiedTableName);
+				assertTrue(resultSet.next(), "Expected exactly 1 row from " + qualifiedTableName + ", got 0");
+				assertTrue(resultSet.getInt(COLUMN_NAME) == 1, "Value for " + COLUMN_NAME + " should be 1");
+				resultSet.close();
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInDataManipulation) {
+					fail(
+						"supportsCatalogsInDataManipulation is true but INSERT in " + qualifiedTableName + " is not allowed." + 
+						ex.getMessage()
+				  );
+					ex.printStackTrace();
+				}
+			}
+			
+			try {
+				statement.execute(
+					"UPDATE " + qualifiedTableName + " SET " + COLUMN_NAME + " = 2 " +
```

Comment:
String.format would be a bit cleaner here I think

Reply:
Hi @Mause , thanks for taking a look! Prior tests are also using this kind of style, I figured that was just good enough for tests, but I agree, it's nicer with try with resources and string formatter.

Let me know if something else needs changing.